iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ShuntXml.java

```diff
@@ -22,67 +22,53 @@
 
     static final String ROOT_ELEMENT_NAME = "shunt";
 
-    private static final String REGULATING_TERMINAL = "regulatingTerminal";
-
     @Override
     protected String getRootElementName() {
         return ROOT_ELEMENT_NAME;
     }
 
     @Override
     protected boolean hasSubElements(ShuntCompensator sc) {
-        return sc != sc.getRegulatingTerminal().getConnectable();
+        return false;
     }
 
     @Override
     protected void writeRootElementAttributes(ShuntCompensator sc, VoltageLevel vl, NetworkXmlWriterContext context) throws XMLStreamException {
         if (ShuntCompensatorModelType.NON_LINEAR.equals(sc.getModelType())) {
-            throw new PowsyblException("Non linear shunt not yet supported");
+            throw new PowsyblException("Non linear shunts are not supported for IIDM-XML version " + context.getVersion().toString(".")
+                    + ". IIDM-XML version should be >= 1.2");
         }
+        IidmXmlUtil.assertMinimumVersionIfNotDefault(sc.isVoltageRegulatorOn(), ROOT_ELEMENT_NAME, "voltageRegulatorOn",
+                IidmXmlUtil.ErrorMessage.NOT_DEFAULT_NOT_SUPPORTED, IidmXmlVersion.V_1_2, context);
+        IidmXmlUtil.assertMinimumVersionIfNotDefault(!Double.isNaN(sc.getTargetV()), ROOT_ELEMENT_NAME, "targetV",
+                IidmXmlUtil.ErrorMessage.NOT_DEFAULT_NOT_SUPPORTED, IidmXmlVersion.V_1_2, context);
+        IidmXmlUtil.assertMinimumVersionIfNotDefault(!Double.isNaN(sc.getTargetDeadband()), ROOT_ELEMENT_NAME, "targetDeadband",
+                IidmXmlUtil.ErrorMessage.NOT_DEFAULT_NOT_SUPPORTED, IidmXmlVersion.V_1_2, context);
+        IidmXmlUtil.assertMinimumVersionIfNotDefault(sc.getRegulatingTerminal().getConnectable() != sc, ROOT_ELEMENT_NAME, "regulatingTerminal",
+                IidmXmlUtil.ErrorMessage.NOT_DEFAULT_NOT_SUPPORTED, IidmXmlVersion.V_1_2, context);
```

Comment:
in IIDM v1.2, we should not use this class anymore?

Reply:
These lines do not check the version, they check if the shunt is regulating/has a target value/has a regulating terminal/etc... and if it does, it throws an exception because it is not supported in serialization in version older than v1.2.

Reply:
Hum... So you check that the Shunt (in V1.2) is compatible with an old version (the new attributes have the default value)? Do I understand well?

Reply:
Yes!